hasjob/templates/campaign_info.html.jinja2

```diff
@@ -1,103 +1,100 @@
-          </div>
+        <p class="button-bar">
+          {%- for action in campaign.actions %}
+            <a href="{{ action.url_for('edit') }}" class="btn btn-sm btn-{{ action.category }}">{% if action.icon %}<i class="fa fa-{{ action.icon }}"></i> {% endif %}{{ action.title }} <span class="badge">{{ action.anonuseractions.count() }}+{{ action.useractions.count() }}</span> {%- if not action.public %} <i class="fa fa-eye-slash"></i>{% endif %}</a>
+          {%- endfor %}
+          <a title="New action" href="{{ campaign.url_for('action_new') }}" class="btn btn-sm btn-default"><i class="fa fa-plus"></i></a>
+        </p>
+      </div>
+      <div class="col-md-3">
+        <div class="list-group">
+          <a class="list-group-item" href="{{ url_for('index', preview=campaign.name) }}">
+            <i class="fa fa-fw fa-eye"></i>&nbsp;&nbsp; Preview this
+          </a>
+          <a class="list-group-item" href="{{ campaign.url_for('edit') }}">
+            <i class="fa fa-fw fa-pencil"></i>&nbsp;&nbsp; Edit this
+          </a>
+          <a class="list-group-item" href="{{ campaign.url_for('delete') }}">
+            <i class="fa fa-fw fa-trash-o"></i>&nbsp;&nbsp; Delete this
+          </a>
+          {% with reach=campaign.estimated_reach() %}{% if reach != none %}<span class="list-group-item"><i class="fa fa-fw fa-bullhorn"></i>&nbsp;&nbsp; Target size: {{ reach }} {{ ngettext("user", "users", reach) }}</span>{% endif %}{% endwith %}
+          <span class="list-group-item"><i class="fa fa-fw fa-info-circle"></i>&nbsp;&nbsp; Views: {{ campaign.anonviews.count() }}+{{ campaign.views.count() }}</span>
         </div>
       </div>
     </div>
-    {%- if interval %}
-    <div class="section">
-      <div id="chart"></div>
-    </div>
-    <div class="section" role="tabpanel">
-      <ul class="nav nav-tabs">
-        {%- for action in campaign.actions if action.type in ('Y', 'N', 'M', 'F') %}
-          <li role="presentation" {%- if loop.first %} class="active" {%- endif %}><a href="#users-{{ action.name }}" aria-controls="users-{{ action.name }}" role="tab" data-toggle="tab">{{ action.title }}</a></li>
-        {%- endfor %}
-      </ul>
-      <div class="tab-content">
-        {%- for action in campaign.actions if action.type in ('Y', 'N', 'M', 'F') %}
-          <div role="tabpanel" class="tab-pane fade {%- if loop.first %} in active {%- endif %}" id="users-{{ action.name }}">
-            <p class="tab-download">
-              <a href="{{ action.url_for('csv') }}"><i class="fa fa-download"></i> Download CSV</a>
-            </p>
-            <table class="table table-responsive">
-              <thead>
+  </div>
+  {%- if interval %}
+  <div class="section">
+    <div id="chart"></div>
+  </div>
+  <div class="section" role="tabpanel">
+    <ul class="nav nav-tabs">
+      {%- for action in campaign.actions if action.type in ('Y', 'N', 'M', 'F') %}
```

Comment:
A helper method could make this cleaner, especially since the action labels hardcoded?

Reply:
That’s old code. The entire template needs an upgrade. Ive only changed the container div here.

Reply:
There’s a pending model migration as well because I made the mistake of using a SQL enum type for the `type` column.